### PR TITLE
Copy prearchive step to GitHub Actions workflow

### PR DESCRIPTION
**Pull request by oseasmoran73**
## Description

Reference to [24183](https://github.com/department-of-veterans-affairs/va.gov-team/issues/24183)

## Testing done

Successful [run](https://github.com/department-of-veterans-affairs/content-build/runs/2749235399?check_suite_focus=true)

## Acceptance criteria
- [x] Prearchive script is called after integration tests in GitHub Actions workflow.
- [x] Asset links match the S3 bucket corresponding to the build environment.

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -254,3 +269,67 @@ jobs:
           message: "Broken links detected!"
           blocks: ${{ steps.get-broken-link-info.outputs.SLACK_BLOCKS }}
           channel_id: "U01SPJ10WBC"
+
+  prearchive:
+    name: Prearchive Optimizations
+    runs-on: self-hosted
+    if: ${{ github.ref == 'refs/heads/master' }}
```
Comment (oseasmoran73) [inline .github/workflows/continuous-integration.yml:276]: Execute prearchive step only on master

File: .github/workflows/continuous-integration.yml [hunk 2]
```diff
@@ -202,6 +205,18 @@ jobs:
         id: get-broken-link-info
         run: node ./script/github-actions/check-broken-links.js ${{ matrix.buildtype }}
 
+      - name: Compress ${{ matrix.buildtype }} build
+        if: ${{ github.ref == 'refs/heads/master' && success() }}
+        run: tar -zcvf ${{ matrix.buildType }}.tar.gz build
+
+      - name: Upload build
+        uses: actions/upload-artifact@v2
+        if: ${{ github.ref == 'refs/heads/master' && success() }}
```
Comment (oseasmoran73) [inline .github/workflows/continuous-integration.yml:214]: Execute storing artifact only on master

File: .github/workflows/continuous-integration.yml [hunk 3]
```diff
@@ -254,3 +269,67 @@ jobs:
...
+            ~/.cache/yarn
+            node_modules
+          key: ${{ steps.get-node-version.outputs.NODE_VERSION }}-${{ hashFiles('yarn.lock') }}
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile --prefer-offline
+        env:
+          YARN_CACHE_FOLDER: ~/.cache/yarn
+
+      - name: Download build artifactory
```
Comment (U-DON) [inline .github/workflows/continuous-integration.yml:325]:
```suggestion
      - name: Download build artifact
```

File: .github/workflows/continuous-integration.yml [hunk 4]
```diff
@@ -254,3 +269,67 @@ jobs:
           message: "Broken links detected!"
           blocks: ${{ steps.get-broken-link-info.outputs.SLACK_BLOCKS }}
           channel_id: "U01SPJ10WBC"
+
+  prearchive:
+    name: Prearchive Optimizations
```
Comment (U-DON) [inline .github/workflows/continuous-integration.yml:274]: Is it actually optimizing stuff or just rewriting URLs to link to the buckets? If there isn't optimization involved, I'd be okay with just calling this "Prearchive".

**Review by U-DON — APPROVED** [review]

**Pull request merged**